### DISCUSSION TITLE
Rendering Text-Decorations

### DISCUSSION THREAD

**Discussion by ncallaway** [grovesNL/glyphon#99] [Q&A]
I'm thinking about rendering text-decorations (like `underline`, `strikethrough`). My understanding is that this isn't something built into glyphon (or cosmic text), which means I'll need to render the decoration myself, which makes sense.

What I'm stuck on is rendering _breaks_ or _gaps_ in the decoration to ensure the legibility of the text. Here's a screenshot of firefox with text-decoration underline. Note that the underline is masked out around the descender so that the descender is fully legible.

![Screenshot 2024-06-07 at 22-03-00 text-decoration - CSS Cascading Style Sheets MDN](https://github.com/grovesNL/glyphon/assets/427497/5cbd3236-d4f7-4b0b-a197-9a6543493255)

Does glyphon or cosmic-text provide me any way to know where the descenders are, so I can render the underline correctly? I'd be happy with, like, a list of boxes somewhere in the CPU, or some ability to render text with an additional 1-px width to a stencil texture.

**Comment by BillyDM** [comment]
It would actually be really nice if glyphon had a built-in way to render underlines, selection highlights, and strike-throughs. In theory this could just be done using triangle meshes.

**Reply by BillyDM** [reply]
Oh and an editor cursor.
